netket/vqs/mc/mc_state/expect.py

```diff
@@ -45,7 +46,7 @@ def get_local_kernel_arguments(vstate: MCState, Ô: Squared):  # noqa: F811
     check_hilbert(vstate.hilbert, Ô.hilbert)
 
     σ = vstate.samples
-    σp, mels = Ô.parent.get_conn_padded(σ)
+    σp, mels = replicate_sharding(Ô.parent.get_conn_padded)(σ)
```

Comment:
what does this do exactly if you have 2 nodes with M devices each?

Reply:
it will callback to the  numba operator M times on each of the 2 nodes on n_samples / (2 M) size inputs.